C/services/common/include/plugin_manager.h

```diff
@@ -25,23 +25,34 @@
  */
 class PluginManager {
 	public:
-                static PluginManager *getInstance();
-        	PLUGIN_HANDLE loadPlugin(const std::string& name, const std::string& type);
-		void unloadPlugin(PLUGIN_HANDLE handle);
-    void *resolveSymbol(PLUGIN_HANDLE handle, const std::string& symbol);
-                PLUGIN_HANDLE findPluginByName(const std::string& name);
-                PLUGIN_HANDLE findPluginByType(const std::string& type);
-                PLUGIN_INFORMATION *getInfo(const PLUGIN_HANDLE);
-                static PluginManager            *instance;
+		static PluginManager *getInstance();
+		PLUGIN_HANDLE	loadPlugin(const std::string& name,
+					   const std::string& type);
+		void		unloadPlugin(PLUGIN_HANDLE handle);
+		void*		resolveSymbol(PLUGIN_HANDLE handle,
+					      const std::string& symbol);
+		PLUGIN_HANDLE	findPluginByName(const std::string& name);
+		PLUGIN_HANDLE	findPluginByType(const std::string& type);
+		PLUGIN_INFORMATION
+				*getInfo(const PLUGIN_HANDLE);
+		void		scanPluginDir(const std::string& type,
```

Comment:
Please call tis something that reflects what we are doing not the mechanism it uses. E.g. getInstalledPlugins